Sources/AblyAssetTrackingPublisher/WorkerQueue/PublisherWorkerFactory.swift

```diff
@@ -0,0 +1,21 @@
+import Foundation
+import AblyAssetTrackingInternal
+
+public class PublisherWorkerFactory: WorkerFactory
+{
+    public typealias PropertiesType = PublisherWorkerQueueProperties
+    public typealias WorkerSpecificationType = PublisherWorkSpecification
+    
+    public init() {
+        
+    }
+    
+    public func createWorker(workerSpecification: PublisherWorkSpecification, logHandler: InternalLogHandler?)
+        -> any Worker<PropertiesType, WorkerSpecificationType> {
+
+            switch (workerSpecification) {
+            case .legacy(callback: let callback, logger: let logger):
```

Comment:
The logger doesn’t need to be in the specification – it's already passed to this method as the `logHandler` argument.

Reply:
Changed :)